### PR TITLE
feat(curriculum): create debugging quiz

### LINKED ISSUE #56208: [HACKTOBERFEST 2024] - Debugging Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Debugging quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-debugging`.

**All quiz questions need to relate to one of these topics:**

- Syntax errors
- logical errors
- type errors
- using developer tools
- debugger statements
- console logging - log(), error(), table(), etc
- try...catch...finally
- conditionals to avoid errors
- throw statement and Error() constructor
- breakpoints, watchers, etc

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by nayanmapara**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56208

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 1]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which JavaScript statement can be used to intentionally pause the execution of your code for debugging purposes?
 
 #### --distractors--
 
-Placeholder distractor 1
+try...catch
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:46]:
```suggestion
`try...catch`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 2]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+console.log()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:50]:
```suggestion
`console.log()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 3]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+alert()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:54]:
```suggestion
`alert()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 4]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+debugger
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:58]:
```suggestion
`debugger`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 5]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which console method can be used to display data as a table?
 
 #### --distractors--
 
-Placeholder distractor 1
+console.log()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:68]:
```suggestion
`console.log()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 6]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+console.warn()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:72]:
```suggestion
`console.warn()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 7]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+console.error()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:76]:
```suggestion
`console.error()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 8]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+console.table()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:80]:
```suggestion
`console.table()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 9]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is the correct syntax for using `try...catch` in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+try { ... } else { ... }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:112]:
```suggestion
`try { ... } else { ... }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 10]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+try { ... } catch { ... } finally { ... }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:116]:
```suggestion
`try { ... } catch { ... } finally { ... }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 11]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+try { ... } error { ... }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:120]:
```suggestion
`try { ... } error { ... }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 12]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+try { ... } catch(error) { ... }
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:124]:
```suggestion
`try { ... } catch(error) { ... }`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 13]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which method would you use to inspect variables or expressions during debugging in developer tools?
 
 #### --distractors--
 
-Placeholder distractor 1
+inspect()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:200]:
```suggestion
`inspect()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 14]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+trace()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:204]:
```suggestion
`trace()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 15]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+console.error()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:208]:
```suggestion
`console.error()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 16]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+console.log()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:212]:
```suggestion
`console.log()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 17]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which tool is commonly used to step through code line by line during debugging?
 
 #### --distractors--
 
-Placeholder distractor 1
+console.log()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:244]:
```suggestion
`console.log()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 18]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+alert()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:248]:
```suggestion
`alert()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 19]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+throw statement
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:252]:
```suggestion
`throw` statement
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 20]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What method could you use to track the call history of a function in the console?
 
 #### --distractors--
 
-Placeholder distractor 1
+console.log()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:354]:
```suggestion
`console.log()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 21]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+console.table()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:358]:
```suggestion
`console.table()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 22]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+console.warn()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:362]:
```suggestion
`console.warn()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 23]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+console.trace()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:366]:
```suggestion
`console.trace()`
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This is a great start! Looks like there's a few things missing code tags.

Also you should never commit directly to your `main` branch when contributing to a repository.

**Comment by nayanmapara** [comment]
> This is a great start! Looks like there's a few things missing code tags.
> 
> Also you should never commit directly to your `main` branch when contributing to a repository.

Thank you for pointing that out! I’ll make sure to remember this moving forward and avoid committing directly to the main branch.

File: curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md [hunk 24]
```diff
@@ -17,439 +17,438 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+The program will continue with default values
 
 #### --answer--
 
-Placeholder answer
-
+An uncaught exception error will occur
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-debugging-javascript/66edd10913f078e7669eca81.md:454]:
```suggestion
An uncaught exception error will occur

```

EOF new line.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
One tiny nit and then this will be ready to go.

**Review by naomi-lgbt — APPROVED** [review]
Great work! This looks good to go!

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @nayanmapara. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

- - -

Note for future - it is recommended to make changes on a separate branch, rather than the `main` branch. See more about it in point 3 at https://contribute.freecodecamp.org/how-to-contribute-to-the-codebase/#contributing-to-the-codebase.

**Comment by nayanmapara** [comment]
Thank you, looking forward to future contributions. 😄 
Will keep it in mind to make changes on a separate branch. 💯

**Pull request merged**